cloud/google/machineactuator.go

```diff
@@ -82,17 +83,17 @@ type GCEClientComputeService interface {
 type GCEClient struct {
 	computeService           GCEClientComputeService
 	gceProviderConfigCodec   *gceconfigv1.GCEProviderConfigCodec
+	kubeadm                  kubeadm.Kubeadm
 	scheme                   *runtime.Scheme
 	codecFactory             *serializer.CodecFactory
-	kubeadmToken             string
 	sshCreds                 SshCreds
 	machineClient            client.MachineInterface
 	machineSetupConfigGetter GCEClientMachineSetupConfigGetter
 }
 
 type MachineActuatorParams struct {
 	ComputeService           GCEClientComputeService
-	KubeadmToken             string
+	Kubeadm                  kubeadm.Kubeadm
```

Comment:
TL;DR I'll change it to your suggestion.

I think this might end up as one of the exceptions to that guideline because there is somewhat likely a chance that more than one client will use this interface and in those sceanarios I've seen the interface get defined in another location / more globally, however that is not happening yet, and we should design for what we know and not for a future that never may materialize!